onadata/apps/api/viewsets/briefcase_api.py

```diff
@@ -103,42 +103,44 @@ def __init__(self, *args, **kwargs):
         ]
 
     def get_object(self):
-        formId = self.request.GET.get('formId', '')
-        id_string = _extract_id_string(formId)
-        uuid = _extract_uuid(formId)
-        username = self.kwargs.get('username')
+        form_id = self.request.GET.get('formId', '')
+        id_string = _extract_id_string(form_id)
+        uuid = _extract_uuid(form_id)
 
-        obj = get_instance_or_404(xform__user__username__iexact=username,
-                                  xform__id_string__exact=id_string,
-                                  uuid=uuid)
+        obj = get_instance_or_404(
+            xform__id_string__exact=id_string,
+            uuid=uuid,
+        )
         self.check_object_permissions(self.request, obj.xform)
 
         return obj
 
     def filter_queryset(self, queryset):
         username = self.kwargs.get('username')
-        if username is None and self.request.user.is_anonymous:
-            # raises a permission denied exception, forces authentication
-            self.permission_denied(self.request)
-
-        if username is not None and self.request.user.is_anonymous:
-            profile = get_object_or_404(
-                UserProfile, user__username=username.lower())
-
-            if profile.require_auth:
+        if username is None:
+            # If no username is specified, the request must be authenticated
+            if self.request.user.is_anonymous:
                 # raises a permission denied exception, forces authentication
                 self.permission_denied(self.request)
             else:
-                queryset = queryset.filter(user=profile.user)
+                # Return all the forms the currently-logged-in user can access,
+                # including those shared by other users
+                queryset = super().filter_queryset(queryset)
         else:
-            queryset = super().filter_queryset(queryset)
+            raise exceptions.PermissionDenied(
+                detail=t(
+                    'Cannot access non secure URL. Remove username from '
+                    'aggregate server URL in configuration settings.'
+                ),
+                code=status.HTTP_403_FORBIDDEN,
```

Comment:
In retrospect, I don't think it's worth reshuffling the `if`s. It might be worth returning a plain-text response instead of JSON if ODK Briefcase even shows anything from the server's response—I don't know if it does. We could just drop the "Cannot access non secure URL" part and tell them to remove the username

Reply:
Briefcase does not expose the back-end response. I removed the message entirely.